### DISCUSSION TITLE
[FEATURE REQ] Support torrent re-announcing with Radarr/Sonarr

### DISCUSSION THREAD

**Discussion by cplieger** [autobrr/autobrr#1676] [Ideas]
When using radarr or sonarr as actions to filter releases, there is no re-announce functionality. This leads to torrents getting stuck as stalled for 60-90 min depending on the tracker..

Solution could be 2 things:
- send torrent info to sonarr/radarr to check if valid, if valid do not send torrent, but send directly to qbittorrent  with appropriate tag for sonarr/radarr to pick up the download.
- send torrent to sonarr/radarr and let them push to qbittorrent. If a connection exists in autobrr with the torrent client, re-announce directly there.

Alternatives are using external scripts like qbittools, but native integration would be cleaner.

**Comment by Flupster** [comment]
I'd love to have this feature as well but I see many hurdles

If I recall getting the infoHash from Sonarr was only recently added (Sonarr doesn't have a long changelog so I couldn't find the exact date) I may be misremembering this, however older instances of Sonarr won't support it.

With Sonarr API we have the ability to find the download clients, but not get their password, they're are asterisked out, so auto adding download clients into autobrr isn't possible.

Sonarr can have multiple download clients, however the release/push API returns the `downloadClientId` so with some mapping it's possible to locate the download clients hostname/port from Sonarr's API and see if the same client exists in autobrr.

**Comment by cplieger** [comment]
This does seem complex. Would a workaround be to completely bypass the ARRs for reannouncing?
1. Prompt the user to add the final downloads clients to autobrr if they use actions with ARR apps. 
2. When a push gets approved through sonarr/radarr autobrr can find the torrent in the added download clients.
3. If found execute the reannounce or other actions in the torrent client

If the download clients have not been added, not a problem use default behaviour, maybe a warning. if they have been added advanced options become available.

**Comment by zze0s** [comment]
This is indeed somewhat complex and that's why it's not yet a feature.

I don't think having something that automatically syncs with the arrs is the way forward even if it would be best from a user perspective.

We have previously discussed possible solutions, like improve actions to have sub actions that can run on different event statuses from it's parent.

So in that case for the arr actions you could then add a sub action to run on approved, and that could then be a new action type called re-announce and you select the client, for example.

How the matching happens we'll see. Hash would ofc be best, but we could match on release name or check for downloads never than 1 min and start the re-announce process on those.

It would still require to have the download client setup in autobrr but I don't see that as much of an issue personally.

The topic of "Global/reusable actions" has been brought up multiple times before, so maybe that is something to do at the same time.

**Comment by cplieger** [comment]
Thanks for sharing. That seems like a very powerful addition overall, especially if it can be made scriptable. Could even be used for other sub actions like cross seeding, but that is beyond this topic.

**Comment by rpatterson** [comment]
The only tracker I've observed, where added torrents don't start downloading until the next client announce, is TorrentLeech. I just noticed that their docs specify a 5 second delay before adding torrents announced in IRC. So I wonder how many of the use cases for re-announces could be much supported in a much more maintainable way by just enriching delay support.

I use one filter for all my indexers. This expresses my intention: "Grab releases from any indexers that support IRC announces, I don't really care much about this indexer vs that one." But more importantly its easy for me to maintain as a user. So I added a 5 second delay to my one filter as a workaround. That's actually fine for me, but a core use case for IRC announcing is to optimize ratio by getting into the swarm early and I could imagine that getting established in the swarm 5 seconds earlier might actually result in a significantly better ratio, so it's sub-optimal to set this delay across all my indexers.

So maybe Autobrr can add support for setting a delay in the individual indexer settings? And even better it could add per-indexer default delays into the indexer definitions built into Autobrr. If Autobrr provided those richer delay settings, how many of the users following this discussion would still need support for re-announcing? I know I would not need it.
